tests/cypress/integration/e2e/collection.spec.js

```diff
@@ -0,0 +1,138 @@
+  context('Filter by Old First', () => {
+    it('Should display the various collections and their details.', () => {
+      cy.get('.is-grouped > :nth-child(1) > .control > .select > select')
+        .then(($select) => {
+          const opt = $select.find('option[value="BLOCK_NUMBER_ASC"]')
+          $select.val(opt.attr('value'))
+          return $select
+        })
+        .trigger('change')
+    })
+    it('Validate Page loaded and view oldest first', () => {
+      cy.get('.columns').should('be.visible')
+      cy.wait(2000)
+    })
+  })
+
+  context('Filter by Last Interacted', () => {
+    it('Should display the various collections and their details.', () => {
+      cy.get('.is-grouped > :nth-child(1) > .control > .select > select')
+        .then(($select) => {
+          const opt = $select.find('option[value="UPDATED_AT_DESC"]')
+          $select.val(opt.attr('value'))
+          return $select
+        })
+        .trigger('change')
+    })
+    it('Validate Page loaded and view Last Interacted', () => {
+      cy.get('.columns').should('be.visible')
+      cy.wait(2000)
+    })
+  })
+  context('Filter by Most Expensive', () => {
+    it('Should display the various collections and their details.', () => {
+      cy.get('.is-grouped > :nth-child(1) > .control > .select > select')
+        .then(($select) => {
+          const opt = $select.find('option[value="PRICE_DESC"]')
+          $select.val(opt.attr('value'))
+          return $select
+        })
+        .trigger('change')
+    })
+    it('Validate Page loaded and view Last Interacted', () => {
+      cy.get('.columns').should('be.visible')
+      // cy.wait(2000)
+    })
+  })
+  context('Filter by from cheaper', () => {
+    it('Should display the various collections and their details.', () => {
+      cy.get('.is-grouped > :nth-child(1) > .control > .select > select')
+        .then(($select) => {
+          const opt = $select.find('option[value="PRICE_ASC"]')
+          $select.val(opt.attr('value'))
+          return $select
+        })
+        .trigger('change')
+    })
+    it('Validate Page loaded and view Last Interacted', () => {
+      cy.get('.columns').should('be.visible')
+      // cy.wait(2000)
+    })
+  })
+  describe('Collection Search Functionality.', () => {
+    it('Serach for a collection', () => {
+      cy.get('.is-grouped > :nth-child(1) > .control > .select > select')
+        .then(($select) => {
+          const opt = $select.find('option[value="BLOCK_NUMBER_DESC"]')
+          $select.val(opt.attr('value'))
+          return $select
+        })
+        .trigger('change')
+      cy.get('.field.is-expanded > .control > .input').type('japandas')
+      cy.wait(2000)
+      cy.get(':nth-child(3) > .columns').should('be.visible')
+      cy.get('.field.is-expanded > .control > .input').clear()
+    })
+  })
+  describe('Collection Buy Now.', () => {
+    it('Click on Buy Now button', () => {
+      cy.get('.mb-5 > .switch > .check').click()
```

Comment:
this is also way to generic of a  selector. try to stick to `data-cy` pls. this goes for all the other generic queries as well